pkg/proc/gdbserial/gdbserver.go

```diff
@@ -1919,8 +1922,16 @@ func (regs *gdbRegisters) Slice(floatingPoint bool) ([]proc.Register, error) {
 
 			value := regs.regs[reginfo.Name].value
 			xmmName := "x" + reginfo.Name[1:]
-			r = proc.AppendBytesRegister(r, strings.ToUpper(xmmName), value[:16])
-			r = proc.AppendBytesRegister(r, strings.ToUpper(reginfo.Name), value[16:])
+			r = proc.AppendBytesRegister(r, strings.ToUpper(xmmName), value)
+
+		case reginfo.Bitsize == 512:
+			if !strings.HasPrefix(strings.ToLower(reginfo.Name), "zmm") || !floatingPoint {
+				continue
+			}
+
+			value := regs.regs[reginfo.Name].value
+			xmmName := "x" + reginfo.Name[1:]
```

Comment:
Shouldn't this technically be `"z" + reginfo.Name[1:]` since we're referring to the full 512-bit width register? Similarly shouldn't above be `"y" + ...` and the one even before be `"x" + ...`? 

xmm refers to lower 128-bits, ymm refers to the 256-bit, and zmm is for AVX-512 referring to the full 512-bit width of the register, correct?

Reply:
DWARF doesn't have separate register numbers for XMMx, YMMx and ZMMx, they all map to XMM. We could change amd64DwarfToName to accept all three variants, but it isn't worth it, and the correct name gets assigned by formatSSEReg later on anyway. This is just an internal thing.

Reply:
Ack, just wanted to make sure. Thanks.